docs: Synthetics monitoring (#209)
elastic/observability-docs: docs/en/observability/synthetics-monitoring.asciidoc

[[synthetics-monitoring]]

> = Synthetics monitoring

Reviewer: s/`synthetics/synthetic/`

++++

> <titleabbrev>Synthetics monitoring</titleabbrev>

Reviewer: `s/synthetics/synthetic/`

++++

[WARNING]
====
This functionality is experimental and may be changed or removed completely in a future release.
Elastic will take a best effort approach to fix any issues, but experimental features are not subject
to the support SLA of official GA features.

Have a question? Want to leave feedback? Visit the
https://discuss.elastic.co/tags/c/observability/uptime/75/synthetics[Synthetics discussion forum].
====

[role="screenshot"]

> image::images/synthetic-app-overview.png[Synthetics app overview]

Reviewer: This image was taken when the `url` links were broken. Could we swap it with a more recent capture that should have that fixed?

Author: Done.

Synthetic monitoring allows you to simulate actions and requests that an end-user would perform
on your site at predefined intervals and in a controlled environment.
The result is rich, consistent, and repeatable data that you can trend and alert on.
For example, test popular user journeys, like logging in, adding items to a cart, and checking
out -- actions that need to work for your users consistently.
Alerting ensures any degraded performance or broken actions are fixed before impacting your
bottom line or customers' experience.

[discrete]
[[how-synthetics-works]]
== How does it work?

While there are different ways to utilize Elastic's synthetic monitoring,
the easiest is with our provided Docker image:

// Operational use case screenshot
image::images/synthetics-overview.png[Synthetics overview]

In this use-case, Docker spins up Heartbeat, which has the `@elastic/synthetics` package installed.
Heartbeat handles the scheduling of test runs and supervises the execution of the
`@elastic/synthetics` package.

> `@elasic/synthetics` runs your tests by invoking the Playwright library and starting a new

Reviewer: `s/elasic/elastic/`

Elastic will take a best effort approach to fix any issues, but experimental features are not subject
to the support SLA of official GA features.

Have a question? Want to leave feedback? Visit the
https://discuss.elastic.co/tags/c/observability/uptime/75/synthetics[Synthetics discussion forum].
====

[role="screenshot"]
image::images/synthetic-app-overview.png[Synthetics app overview]

Synthetic monitoring allows you to simulate user actions and requests at predefined intervals
and in a controlled environment.
The result is rich, consistent, and repeatable data that you can trend and alert on.
For example, test popular user journeys, like logging in, adding items to a cart, and checking
out -- actions that need to work for your users consistently.
Alerting ensures any degraded performance or broken actions are fixed before impacting your
bottom line or customers' experience.

[discrete]
[[how-synthetics-works]]
== How does it work?

While there are different ways to utilize Elastic's synthetic monitoring,
the easiest is with our provided Docker image:

// Operational use case screenshot
image::images/synthetics-overview.png[Synthetics overview]

> In this use-case, Docker spins up Heartbeat and the `@elastic/synthetics` package.

Reviewer 2: ```suggestion
In this use-case, Docker spins up Heartbeat which has `@elastic/synthetics` package installed.
```

Heartbeat handles the scheduling of test runs and supervises the execution of the
`@elastic/synthetics` package.
`@elasic/synthetics` runs your tests by invoking the Playwright library and starting a new
browser process.
Performance data is then gathered and reported back to Heartbeat,
where it is formatted and sent to the Elastic Stack.
At this point, the data is available in the Uptime app to view, analyze, and alert on.

// REVIEWERS

> // Do we need a separate image for the --> suite of tests in user-controlled git repo?

Reviewer 2: I dont think at this point.
